api/types/registry/registry.go

```diff
@@ -102,3 +105,15 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// DistributionInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+type DistributionInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
+	// ID is the local image digest
+	ID digest.Digest
```

Comment:
The `ID` isn't useful immediately, but we have discussed supporting local images on Swarm workers using image IDs instead of pulling them by tag/digest. You're right that `ID` doesn't make sense for manifest lists, so this is limited to only `schema2` images. Does adding an `omitempty` tag set as @aaronlehmann suggested avoid that issue?